File: tests/fixtures/realistic/gin_chain/store.go
// Sink helper for the gin_chain fixture.
//
// runQuery() takes a parameter and passes it into a SQL sink.
// Cross-file summary should record params_to_sink for param 0
// with rule go/taint-sql-injection.

package gin_chain

import "database/sql"

var db *sql.DB

func runQuery(term string) []string {
	_, _ = db.Query("SELECT * FROM users WHERE name = '" + term + "'")
	return nil
}

File: tests/fixtures/realistic/js_multihop/store.js
// Sink helper (fileC) for the js_multihop fixture.
//
// runQuery() concatenates its parameter into a SQL string and passes it to a
// db.query sink. Its single-file summary records params_to_sink for param 0
// with rule js/taint-sql-injection. Scanned alone, `term` is just a parameter
// (not a source), so no taint finding fires here.

const db = { query(_q) { return []; } };

function runQuery(term) {
    return db.query("SELECT * FROM users WHERE name = '" + term + "'");
}

module.exports = { runQuery };
